File: 0202-happy-number/0202-happy-number.java
class Solution {
    public boolean isHappy(int n) {
        while(n!=1 && n!=4){
            n = getSumOfSquares(n);
        }

        return n==1;
    }

    public int getSumOfSquares(int n){
        int sum = 0;
        while(n>0){
            int digit = n%10;
            n/=10;
            sum+=digit*digit;
        }

        return sum;
    }
}

File: 0202-happy-number/0202-happy-number.cpp
class Solution {
public:
    bool helper(int n,set<int>& used){
        if(n == 1) return true;
        if(used.count(n)) return false;

        used.insert(n);
        int sum = 0;
        while(n){
            sum += (n%10)*(n%10);
            n/=10;
        }
        return helper(sum,used);
    }

    bool isHappy(int n) {
        set<int> used;
        return helper(n,used);
    }
};